include/fuzzuf/cli/fuzzer/aflplusplus/build_aflplusplus_fuzzer_from_args.hpp

```diff
@@ -0,0 +1,284 @@
+      /* dumb_mode */ false,  // FIXME: add dumb_mode
+      fuzzuf::utils::CPUID_BIND_WHICHEVER, schedule,
+      aflplusplus_options.schedule);
+
+  // NativeLinuxExecutor needs the directory specified by "out_dir" to be
+  // already set up so we need to create the directory first, and then
+  // initialize Executor
+  fuzzuf::utils::SetupDirs(setting->out_dir.string());
+
+  using fuzzuf::algorithm::afl::option::GetDefaultOutfile;
+  using fuzzuf::algorithm::afl::option::GetMapSize;
+  using fuzzuf::cli::ExecutorKind;
+
+  std::shared_ptr<TExecutor> executor;
+  switch (global_options.executor) {
+    case ExecutorKind::NATIVE: {
+      auto nle = std::make_shared<fuzzuf::executor::NativeLinuxExecutor>(
+          setting->argv, setting->exec_timelimit_ms, setting->exec_memlimit,
+          setting->forksrv,
+          setting->out_dir / GetDefaultOutfile<AFLplusplusTag>(),
+          GetMapSize<AFLplusplusTag>(),  // afl_shm_size
+          0                              // bb_shm_size
+      );
+      executor = std::make_shared<TExecutor>(std::move(nle));
+      break;
+    }
+
+    case ExecutorKind::FORKSERVER: {
+      auto lfe = std::make_shared<fuzzuf::executor::LinuxForkServerExecutor>(
+          fuzzuf::executor::LinuxForkServerExecutorParameters()
+              .set_argv(setting->argv)
+              .set_exec_timelimit_ms(setting->exec_timelimit_ms)
+              .set_exec_memlimit(setting->exec_memlimit)
+              .set_path_to_write_input(setting->out_dir /
+                                       GetDefaultOutfile<AFLplusplusTag>())
+              .set_afl_shm_size(GetMapSize<AFLplusplusTag>())  // afl_shm_size
+              .move());
+      executor = std::make_shared<TExecutor>(std::move(lfe));
+      break;
+    }
+
+    case ExecutorKind::QEMU: {
+      // NOTE: Assuming GetMapSize<AFLplusplusTag>() ==
+      // QEMUExecutor::QEMU_SHM_SIZE
+      auto qe = std::make_shared<fuzzuf::executor::QEMUExecutor>(
+          global_options.proxy_path.value(), setting->argv,
+          setting->exec_timelimit_ms, setting->exec_memlimit, setting->forksrv,
+          setting->out_dir / GetDefaultOutfile<AFLplusplusTag>());
+      executor = std::make_shared<TExecutor>(std::move(qe));
+      break;
+    }
+
+#ifdef __aarch64__
+    case ExecutorKind::CORESIGHT: {
+      auto cse = std::make_shared<fuzzuf::executor::CoreSightExecutor>(
+          global_options.proxy_path.value(), setting->argv,
+          setting->exec_timelimit_ms, setting->exec_memlimit, setting->forksrv,
+          setting->out_dir / GetDefaultOutfile<AFLplusplusTag>(),
+          GetMapSize<AFLplusplusTag>()  // afl_shm_size
+      );
+      executor = std::make_shared<TExecutor>(std::move(cse));
+      break;
+    }
+#endif
+
+    default:
+      EXIT("Unsupported executor: '%s'", global_options.executor.c_str());
+  }
+
+  auto mutop_optimizer = std::unique_ptr<optimizer::Optimizer<u32>>(
+      new algorithm::afl::AFLHavocCaseDistrib());
```

Comment:
Is it no problem that this is not `AFLplusplusHavocCaseDistrib`?

Reply:
fixed, thank you for pointing it out